### ISSUE TITLE
How to Correctly Model Array of Enums

### ISSUE THREAD

**Issue by kevinconaway** [springfox/springfox#810]
How can I correctly model an array of enum values?

If I have the following:

```java
@ApiModel(value="A permission")
public enum Permission {
    @ApiModelProperty(value="Can read")
    read,
    @ApiModelProperty(value="Can write")
    write
}
```
and

```java
@ApiModel(value="Some resource"
public class MyResource {

    @ApiModelProperty(value="Permissions to apply")
     private Set<Permission> permissions;
}
```

I would expect that `MyResource` would correctly show the _permissions_ property to be an array with the _allowableValues_ set to the values of the enum.  However, the field is rendered as:

```javascript
permissions: {
    type: "array",
    description: "Permissions to apply",
    items: {
        type: "string"
    }
}
```

How should this be rendered?

**Comment by dilipkrish** [comment]
Closing this as its a duplicate of #801. This is because response models do not support allowable values yet.

**Comment by pbeaudouinexpedia** [comment]
My understanding of 06a33f0 is that allowable values are automatically documented based on the enum.

However, the example given by @kevinconaway still does nothing (no enum or enum field documentation).

Maven dependency tree:
> [INFO] +- io.springfox:springfox-swagger2:jar:2.9.0:compile
> [INFO] |  +- io.swagger:swagger-annotations:jar:1.5.18:compile
> [INFO] |  +- io.swagger:swagger-models:jar:1.5.18:compile
> [INFO] |  +- io.springfox:springfox-swagger-common:jar:2.9.0:compile
> [INFO] +- io.springfox:springfox-swagger-ui:jar:2.9.0:compile

Are we using the annotations incorrectly?

**Comment by dilipkrish** [comment]
@pbeaudouinexpedia this is a very old issue. Could you perhaps give an example of what you're situation is? Your model and what you're seeing rendered in the spec?

**Comment by pbeaudouinexpedia** [comment]
@dilipkrish I'm trying to document an enum with `@ApiModel` just like I am documenting a class. I also would like to document the enum values using `@ApiModelProperty` just like it works with class fields.

At the moment the workaround is to document the enum every time it is passed as a parameter, which isn't maintainable.

The example given in the first comment, https://github.com/springfox/springfox/issues/810#issue-90381700 is exactly what I am trying to do.

When I do this, the text entered in the `@ApiModel` and `@ApiModelProperty` annotations is not visible in Swagger UI. When I do the same on a class and class fields, the text is visible in Swagger UI.

I looked at creating a unit test to demonstrate the issue but I am not sure where to start. I found a couple of candidates:
- ApiModelBuilderSpec
- FieldModelPropertyProviderSpec from springfox-swagger1, that file got deleted
